flow/instrumentation.h

```diff
@@ -16,14 +16,35 @@ namespace flutter {
 
 class Stopwatch {
  public:
-  explicit Stopwatch(fml::Milliseconds frame_budget = fml::kDefaultFrameBudget);
+  /// The update type that is passed to the Stopwatch's constructor.
+  enum FrameBudgetUpdateType {
+    /// Update the latest frame budget value from refresh rate everytime.
+    /// See: `DisplayManager::GetMainDisplayRefreshRate`
+    kUpdateEverytime,
+    /// Using the default value `fml::kDefaultFrameBudget` (60 fps)
+    kOnShotValue,
+  };
+
+  /// The delegate interface for `Stopwatch`.
+  class Delegate {
+   public:
+    /// Time limit for a smooth frame.
+    /// See: `DisplayManager::GetMainDisplayRefreshRate`.
+    virtual fml::Milliseconds GetFrameBudget() = 0;
+  };
+
+  /// The constructor to save the on-shot initial_frame_budget value.
+  explicit Stopwatch(
+      fml::Milliseconds initial_frame_budget = fml::kDefaultFrameBudget);
+
+  /// The constructor with a delegate parameter, it will update frame_budget
+  /// everytime when `GetFrameBudget()` is called.
+  explicit Stopwatch(Delegate* delegate);
```

Comment:
Please avoid using raw pointers like this, it makes the ownership model very unclear.

Typically for delegates like this we end up using references.

Reply:
Done~